### PR TITLE
Documentation for sbt-datatype

### PR DESCRIPTION
**Pull request by Duhemm**

### CODE REVIEW

**Comment by dwijnand** [comment]
Cool, TIL a bunch. Looks good to my noob eyes.

**Comment by Duhemm** [comment]
@dwijnand Thanks! I hope there are not too many grammar errors?

**Comment by dwijnand** [comment]
Didn't spot any.

File: src/reference/03-Developers-Guide/06-Datatype/00.md [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+out: Datatype.html
+---
+
+Datatype
+--------
+
+datatype is a code generation library and an sbt autoplugin that generates
+growable datatypes and helps developers avoid breakage of binary compatibility.
+
+It takes a datatype definition in the form of a `JSON` object and generates the
```
Comment (eed3si9n) [inline src/reference/03-Developers-Guide/06-Datatype/00.md]: We should mention that the JSON format is based on Apache Avro.

File: src/reference/03-Developers-Guide/06-Datatype/00.md [hunk 2]
```diff
@@ -0,0 +1,494 @@
+---
+out: Datatype.html
+---
+
+Datatype
+--------
+
+datatype is a code generation library and an sbt autoplugin that generates
+growable datatypes and helps developers avoid breakage of binary compatibility.
```
Comment (eed3si9n) [inline src/reference/03-Developers-Guide/06-Datatype/00.md]: We should expand on what we mean by "growable," contrasting the datatypes generated by Datatype with plain case classes.

File: src/reference/03-Developers-Guide/06-Datatype/00.md (hunk 1 above)
Comment (Duhemm) [inline src/reference/03-Developers-Guide/06-Datatype/00.md]: Done

File: src/reference/03-Developers-Guide/06-Datatype/00.md (hunk 2 above)
Comment (Duhemm) [inline src/reference/03-Developers-Guide/06-Datatype/00.md]: Done

**Pull request merged**